File: 0922-sort-array-by-parity-ii/0922-sort-array-by-parity-ii.py
class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:
        odds, evens = [], []
        for n in nums:
            if n % 2 == 0:
                evens.append(n)
            else:
                odds.append(n)
        i, j, ans = 0, 0, []
        while i < len(odds) or j < len(odds):
            ans += [evens[j]] + [odds[i]]
            i += 1
            j += 1
        while i < len(odds):
            ans += [odds[i]]
            i += 1
        while j < len(evens):
            ans += [evens[j]]
            j += 1
        return ans
                

File: 0922-sort-array-by-parity-ii/0922-sort-array-by-parity-ii.ts
function sortArrayByParityII(nums: number[]): number[] {
    let i = 0, j = 1
    while (i < nums.length && j < nums.length) {
        if (nums[i] % 2 === 0) {
            i += 2
        } else if (nums[j] % 2 !== 0) {
            j += 2
        } else {
            [nums[j], nums[i]] = [nums[i], nums[j]]
            i += 2
            j += 2
        }
    }
    return nums
};